src/libraries/System.Runtime.Loader/ref/System.Runtime.Loader.cs

```diff
@@ -12,6 +12,12 @@ public static partial class AssemblyExtensions
         public unsafe static bool TryGetRawMetadata(this System.Reflection.Assembly assembly, out byte* blob, out int length) { throw null; }
         public static void ApplyUpdate(Assembly assembly, ReadOnlySpan<byte> metadataDelta, ReadOnlySpan<byte> ilDelta, ReadOnlySpan<byte> pdbDelta) { throw null; }
     }
+    public static partial class MetadataUpdater
+    {
+        public static void ApplyUpdate(Assembly assembly, ReadOnlySpan<byte> metadataDelta, ReadOnlySpan<byte> ilDelta, ReadOnlySpan<byte> pdbDelta) { throw null; }
```

Comment:
My original reason for having this API on AssemblyExtensions was to use it to implement modern equivalent of Reflection.Emit eventually. This new API shape does not make sense for the eventual Reflection.Emit use case.
Reflection.Emit assemblies would always support updating, independent on whether the debugger is attached, etc.

Just pointing it out. I can live with this shape. If/once we do something to modernize Reflection.Emit, we can introduce a different ApplyUpdate method in the right place.

Reply:
The API review committee ask for the move to MetadataUpdater class:  https://github.com/dotnet/runtime/issues/51159#issuecomment-866199485